### ISSUE TITLE
Ingesters report bad error when tag is rejected

### ISSUE THREAD

**Issue by traetox** [gravwell/gravwell#4]
What I did:
1. Fired up a federator with a fairly restrictive globbing pattern:
```
Tags=test
Tags=testX*
```
2. Fired up a generator and attempted to push with the tag `foobar`

What happened:
1. Generator was rightfully rejected from the federator
2. Error was misleading, indicating a timeout
```
2020/05/26 16:30:31 ERROR: Timed out waiting for active connection due to Connection timeout
``` 
What I expected:
1. Error indicates that the tag negotiation failed

Additional notes:
1. I was using the regex generator from commit `f8d3f73b72a3084db9911ae2b1a187ec915f05c2`

**Comment by djfritz** [comment]
@traetox see muxer.go:516 - we don't ever fail on a failed tag negotiation, we just bounce the connection and skip that indexer. Changing that changes some larger behaviors, so we may need to think about the logic here. Options:

- we log the negotiation failure and just allow WaitForHot to fail with a timeout
- we change the larger behavior and bail with an error if any indexers fail to negotiate

**Comment by traetox** [comment]
I THINK the right answer is to bail when we fail to negotiate.  There are two cases where the negotiation phase will fail (outside of outright failure):

1. Tags are rejected due to federator or malformed
2. Authentication is bad

Both scenarios are fatal and should probably cause the ingester to stop.  Tags getting rejected after the initial handshake mean they were dynamic tags, so the connection bounce is probably the most appropriate action.

Either way, we need SOMETHING to indicate why it failed, i think a log message as well as a message out to standard error is in order.

**Comment by djfritz** [comment]
Fixed by #26
